app/main.py

```diff
@@ -1 +1,34 @@
-# write your code here
+class Animal:
+    alive = []
+
+    def __init__(
+            self,
+            name: str,
+            health: int = 100,
+            hidden: bool = False
+    ) -> None:
+        self.name = name
+        self.health = health
+        self.hidden = hidden
+        Animal.alive.append(self)
+
+    def __repr__(self) -> str:
+        return f"{{Name: {self.name}," \
+               f" Health: {self.health}, Hidden: {self.hidden}}}"
+
+
+class Herbivore(Animal):
+    def hide(self) -> None:
+        if self.hidden is True:
+            self.hidden = False
+        else:
+            self.hidden = True
+
+
+class Carnivore(Animal):
+    @classmethod
+    def bite(cls, animal: Animal) -> None:
+        if animal.hidden is False and not isinstance(animal, Carnivore):
+            animal.health -= 50
+        if animal.health <= 0:
+            super().alive.remove(animal)
```

Comment:
The line `super().alive.remove(animal)` is incorrect. `super()` is used to call methods from a parent class, but here you are trying to access a class variable. You should use `Animal.alive.remove(animal)` instead to correctly access the `alive` list.